internal/provider/data_source_secrets.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/juju/terraform-provider-juju/internal/juju"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSourceWithConfigure = &secretDataSource{}
+
+func NewSecretDataSource() datasource.DataSource {
+	return &secretDataSource{}
+}
+
+type secretDataSource struct {
+	client *juju.Client
+
+	// context for the logging subsystem.
+	subCtx context.Context
+}
+
+// secretDataSourceModel is the juju data stored by terraform.
+// tfsdk must match secret data source schema attribute names.
+type secretDataSourceModel struct {
+	// Model to which the secret belongs.
+	Model types.String `tfsdk:"model"`
+	// Name of the secret to be updated or removed.
```

Comment:
s/to be updated or removed//.

Those words are not accurate.

Reply:
Removed.